keras/models.py

```diff
@@ -1432,7 +1438,15 @@ def input_validation(generator_output):
                 raise Exception('All input arrays and target arrays must have '
                                 'the same number of samples.')
             return data, sample_weight
-
+        if do_validation:
+            data_val, sample_weight_val = input_validation(validation_data)
+            sample_weight_val_l = [standardize_weights(data_val[name],
```

Comment:
You could standardize the weights in `input_validation`, return a dict, then turn the dict into a list afterwards. That way you still have the dict around for `evaluate`, but weight standardization is being done in just one place.